App/Models/SyncSessionModel.cs

```diff
@@ -0,0 +1,249 @@
+    {
+        var str =
+            $"{linePrefix}{FriendlyByteConverter.FriendlyBytes(SizeBytes)}\n" +
+            $"{linePrefix}{FileCount:N0} files\n" +
+            $"{linePrefix}{DirCount:N0} directories";
+        if (SymlinkCount > 0) str += $"\n{linePrefix} {SymlinkCount:N0} symlinks";
+
+        return str;
+    }
+}
+
+public class SyncSessionModel
+{
+    public readonly string Identifier;
+    public readonly string Name;
+
+    public readonly string LocalPath = "Unknown";
+    public readonly string RemoteName = "Unknown";
+    public readonly string RemotePath = "Unknown";
+
+    public readonly SyncSessionStatusCategory StatusCategory;
+    public readonly string StatusString;
+    public readonly string StatusDescription;
+
+    public readonly SyncSessionModelEndpointSize LocalSize;
+    public readonly SyncSessionModelEndpointSize RemoteSize;
+
+    public readonly string[] Errors = [];
+
+    public string StatusDetails
+    {
+        get
+        {
+            var str = $"{StatusString} ({StatusCategory})\n\n{StatusDescription}";
+            foreach (var err in Errors) str += $"\n\n{err}";
+            return str;
+        }
+    }
+
+    public string SizeDetails
+    {
+        get
+        {
+            var str = "Local:\n" + LocalSize.Description("  ") + "\n\n" +
+                      "Remote:\n" + RemoteSize.Description("  ");
+            return str;
+        }
+    }
+
+    // TODO: remove once we process sessions from the mutagen RPC
+    public SyncSessionModel(string localPath, string remoteName, string remotePath,
+        SyncSessionStatusCategory statusCategory,
+        string statusString, string statusDescription, string[] errors)
+    {
+        Identifier = "TODO";
+        Name = "TODO";
+
+        LocalPath = localPath;
+        RemoteName = remoteName;
+        RemotePath = remotePath;
+        StatusCategory = statusCategory;
+        StatusString = statusString;
+        StatusDescription = statusDescription;
+        LocalSize = new SyncSessionModelEndpointSize
+        {
+            SizeBytes = (ulong)new Random().Next(0, 1000000000),
+            FileCount = (ulong)new Random().Next(0, 10000),
+            DirCount = (ulong)new Random().Next(0, 10000),
+        };
+        RemoteSize = new SyncSessionModelEndpointSize
+        {
+            SizeBytes = (ulong)new Random().Next(0, 1000000000),
+            FileCount = (ulong)new Random().Next(0, 10000),
+            DirCount = (ulong)new Random().Next(0, 10000),
+        };
+
+        Errors = errors;
+    }
+
+    public SyncSessionModel(State state)
+    {
+        Identifier = state.Session.Identifier;
+        Name = state.Session.Name;
+
+        // If the protocol isn't what we expect for alpha or beta, show
+        // "unknown".
+        if (state.Session.Alpha.Protocol == Protocol.Local && !string.IsNullOrWhiteSpace(state.Session.Alpha.Path))
+            LocalPath = state.Session.Alpha.Path;
```

Comment:
Looks like we're still assuming alpha is always local and beta is always remote.